pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Agent.groovy

```diff
@@ -49,11 +49,18 @@ import javax.annotation.CheckForNull
 @ToString
 @EqualsAndHashCode
 @SuppressFBWarnings(value="SE_NO_SERIALVERSIONID")
-class Agent extends MappedClosure<Object,Agent> implements Serializable {
+class Agent implements Serializable {
```

Comment:
I'm not quite sure how the serialization is used here, but do we need to worry about compatibility with the old version of the class? If so, there will be issues because of the class hierarchy change and implicit `serialVersionUID`.

Reply:
Hmmm...the only way we could have issues would be with currently running builds at upgrade time. I can probably work around that, actually. Lemme see what I can do.

Reply:
Think I got it. =)

Reply:
I think that adding fields and constructors will change the computed `serialVersionUID` so there will still be problems (unless the serialization library being used ignores `serialVersionUID`). You'd need to check what the old computed value was and add its value explicitly here to avoid that issue.

Reply:
And FWIW I ran the following in the Jenkins script console with Declarative 1.3.2 installed and it looks like the `serialVersionUID` of the class before the change is `-9134086848416921688`:

    println(ObjectStreamClass.lookup(org.jenkinsci.plugins.pipeline.modeldefinition.model.Agent.class).getSerialVersionUID())